ipn/local.go

```diff
@@ -928,7 +928,9 @@ func (b *LocalBackend) enterState(newState State) {
 		b.send(Notify{State: &newState})
 	}
 
+	b.mu.Lock()
 	b.state = newState
```

Comment:
can this be done in the locked section above or does it need to happen after the `b.send`?

If we're re-acquiring the lock, can the state have changed meanwhile? Should it compare c.state to state again before overwriting it?

Reply:
After looking around, it seems like `b.send` actually doesn't need to happen first (though I was a bit hesitant to change order, given that it was specifically written like this).